From julia@diku.dk  Sat Jul 11 11:51:47 2009
From: Julia Lawall <julia@diku.dk>
Date: Sat, 11 Jul 2009 10:32:23 +0200 (CEST)
Subject: Staging: otus: Drop an unnecessary NULL test
To: jirislaby@gmail.com, gregkh@suse.de, <devel@driverdev.osuosl.org>
Message-ID: <Pine.LNX.4.64.0907111031480.29592@ask.diku.dk>


From: Julia Lawall <julia@diku.dk>

Apparently, the NULL test is not necessary at this point.

The semantic patch that finds this problem is as follows:
(http://www.emn.fr/x-info/coccinelle/)

// <smpl>
@@
type T;
expression E;
identifier i,fld;
statement S;
@@

- T i = E->fld;
+ T i;
  ... when != E
      when != i
  if (E == NULL) S
+ i = E->fld;
// </smpl>

Signed-off-by: Julia Lawall <julia@diku.dk>
Acked-by: Jiri Slaby <jirislaby@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/otus/wwrap.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/staging/otus/wwrap.c
+++ b/drivers/staging/otus/wwrap.c
@@ -1015,11 +1015,6 @@ void kevent(struct work_struct *work)
                container_of(work, struct usbdrv_private, kevent);
     zdev_t *dev = macp->device;
 
-    if (macp == NULL)
-    {
-        return;
-    }
-
     if (test_and_set_bit(0, (void *)&smp_kevent_Lock))
     {
         //schedule_work(&macp->kevent);
